### ISSUE TITLE
Generalise / consolidate voting logic

### ISSUE THREAD

**Issue by maurolacy** [confio/poe-contracts#194]
Currently we have a bunch of similar but slightly different implementation of the voting logic.

Moreover: we have a `voting-contract` package that exists to supposedly avoid that.

It'll be good to move the `voting-contract` package to `cw-plus` at some point (or to its own repository), and change all multisigs (the ones in `cw-plus`, `poe-contracts`, and `tgrade-contracts`) to use it.

**Comment by ueco-jb** [comment]
cc: @webmaster128 @uint

**Comment by uint** [comment]
Makes sense. I feel like it'd more be that we'd make the `cw3` spec/package properly reusable/extensible, and use it everywhere. As long as it doesn't break the spec.

Should this ticket be in `cw-plus`?
